### ISSUE TITLE
covariant does not always propagate down

### ISSUE THREAD

**Issue by mraleph** [dart-lang/sdk#31037]
```dart
class Base {
  set f(covariant Object x) { }
}

class Middle extends Base {
  int f;
}

class Child extends Middle {
}
```

When this is compiled with fasta I get (stripping away constructors):

```
  class Base extends core::Object {
    set f(covariant core::Object x) → void {}
  }
  class Middle extends z::Base {
    field core::int f = null;
  }
  class Child extends z::Middle {
  }
```

However if I remove `class Child` and compile again then I get:

```
  class Base extends core::Object {
    set f(covariant core::Object x) → void {}
  }
  class Middle extends z::Base {
    covariant field core::int f = null;
  }
```

Notice that `Middle.f` is now marked as a covariant (which what I would expect based on my reading of the informal spec).

**Comment by eernstg** [comment]
Looking at this issue, I discovered that [covariant-overrides.md](https://github.com/dart-lang/sdk/blob/master/docs/language/informal/covariant-overrides.md) is actually silent on the meaning of covariant fields. I submitted a [CL](https://dart-review.googlesource.com/c/sdk/+/12296) to fix that.

**Comment by stereotype441** [comment]
Fixed in 655d3ce665515289b7ff88fccf855a6e5b95b0a5
